### ISSUE TITLE
Could you share the code parameters (such as iterations for adversarial training) when using CIFAR-10 or SYTHIA as dataset?

### ISSUE THREAD

**Issue by HaoyouDeng** [joffery/M-ADA#9]
Hi Fengchun,

Thanks for the codes. When I change the dataset, such as CIFAR-10 or miniImagenet, I found the aug picture is almost the same as the source picture. After some attempts, there is no good results. So could you share the code parameters (such as iterations for adversarial training) when using CIFAR-10 or SYTHIA as dataset?

Thank you so much!

Best,
Haoyou

**Comment by txsing** [comment]
I met the same problem, did you find the parameters for reproducing the results on Cifar10?

**Comment by joffery** [comment]
Hi @HaoyouDeng and @txsing ,

Thanks for your interests in our work!
This is for CIFAR10-C.

`parser = argparse.ArgumentParser(description='PyTorch WideResNet Training')
parser.add_argument('--dataset', default='cifar10', type=str,
                    help='dataset (cifar10 [default] or cifar100)')
parser.add_argument('--epochs', default=200, type=int,
                    help='number of total epochs to run')
parser.add_argument('--epochs_vae', default=200, type=int,
                    help='number of total epochs to run')
parser.add_argument('--start-epoch', default=0, type=int,
                    help='manual epoch number (useful on restarts)')
parser.add_argument('-b', '--batch-size', default=128, type=int,
                    help='mini-batch size (default: 128)')
parser.add_argument('--lr', '--min-learning-rate', default=0.1, type=float,
                    help='initial learning rate')
parser.add_argument('--lr_max', '--adv-learning-rate', default=20, type=float,
                    help='adversarial learning rate')

parser.add_argument('--gamma', default=1, type=float,
                    help='to make them closer in latent space, less gamma, larger distance')
parser.add_argument('--gamma_vae', default=10, type=float,
                    help='balance vae ce and feature loss')
parser.add_argument('--T_adv', default=15, type=int,
                    help='max iterations')
parser.add_argument('--advstart_epoch', default=0, type=int,
                    help='manual epoch number (useful on restarts)')

parser.add_argument('--K', default=2, type=int,
                    help='num of augmented test domains')
parser.add_argument('--T_min', default=10, type=int,
                    help='min epoches')

parser.add_argument('--momentum', default=0.9, type=float, help='momentum')
parser.add_argument('--nesterov', default=True, type=bool, help='nesterov momentum')
parser.add_argument('--weight-decay', '--wd', default=5e-4, type=float,
                    help='weight decay (default: 5e-4)')
parser.add_argument('--print-freq', '-p', default=50, type=int,
                    help='print frequency (default: 10)')
parser.add_argument('--layers', default=16, type=int,
                    help='total number of layers (default: 28)')
parser.add_argument('--widen-factor', default=4, type=int,
                    help='widen factor (default: 10)')
parser.add_argument('--droprate', default=0, type=float,
                    help='dropout probability (default: 0.0)')

parser.add_argument('--augment',action='store_true',
                    help='whether to use standard augmentation (default: True)')
parser.add_argument('--meta',action='store_true',
                    help='whether to use meta learning (default: True)')
parser.add_argument('--innerlrfixed',action='store_true',
                    help='whether to fix the inner learning rate')
parser.add_argument('--add',action='store_true',
                    help='whether to add meta-train and meta-test (default: True)')

parser.add_argument('--resume', default='', type=str,
                    help='p ath to latest checkpoint (default: none)')
parser.add_argument('--name', default='exp2cifar', type=str,
                    help='name of experiment: ')

parser.add_argument('--mode',  default='test', type=str,
                    help='train or test')
parser.add_argument('--tensorboard',  default=True, type=bool,
                    help='Log progress to TensorBoard')

parser.add_argument('--GPU_ID', default=1, type=int,
                    help='GPU_id')
parser.add_argument('--num_updates', default=1, type=int,
                    help='number of meta-train')
parser.add_argument('--epsilon', default=0.1, type=float, help='pgd')

parser.add_argument('--r', default=1, type=int,
                    help='repeat No')`

This is for SYTHIA.
`parser = argparse.ArgumentParser(description='Training')
parser.add_argument('--train_dataset', default='01-DAWN', type=str,
                    help='')
parser.add_argument('--val_dataset', default='01-DAWN', type=str,
                    help='')

parser.add_argument('--epochs', default=50, type=int,
                    help='number of total epochs to run')
parser.add_argument('--epochs_vae', default=200, type=int,
                    help='number of total epochs to run')
parser.add_argument('--start-epoch', default=0, type=int,
                    help='manual epoch number (useful on restarts)')
parser.add_argument('-b', '--batch-size', default=8, type=int,
                    help='mini-batch size (default: 128)')
parser.add_argument('--lr', '--min-learning-rate', default=0.0001, type=float,
                    help='initial learning rate')
parser.add_argument('--lr_max', '--adv-learning-rate', default=2, type=float,
                    help='adversarial learning rate')

parser.add_argument('--gamma', default=1, type=float,
                    help='to make them closer in latent space, less gamma, larger distance')
parser.add_argument('--gamma_vae', default=1000, type=float,
                    help='balance vae ce and feature loss')
parser.add_argument('--T_adv', default=50, type=int,
                    help='max iterations')
parser.add_argument('--advstart_epoch', default=0, type=int,
                    help='manual epoch number (useful on restarts)')

parser.add_argument('--K', default=1, type=int,
                    help='num of augmented test domains')
parser.add_argument('--T_min', default=5, type=int,
                    help='min epoches')

parser.add_argument('--momentum', default=0.9, type=float, help='momentum')
parser.add_argument('--nesterov', default=True, type=bool, help='nesterov momentum')
parser.add_argument('--weight-decay', '--wd', default=5e-4, type=float,
                    help='weight decay (default: 5e-4)')
parser.add_argument('--print-freq', '-p', default=50, type=int,
                    help='print frequency (default: 10)')
parser.add_argument('--layers', default=16, type=int,
                    help='total number of layers (default: 28)')
parser.add_argument('--widen-factor', default=4, type=int,
                    help='widen factor (default: 10)')
parser.add_argument('--droprate', default=0, type=float,
                    help='dropout probability (default: 0.0)')

parser.add_argument('--resume', default='', type=str,
                    help='p ath to latest checkpoint (default: none)')
parser.add_argument('--name', default='FCN50', type=str,
                    help='name of experiment: ')

parser.add_argument('--mode',  default='test', type=str,
                    help='train or test')
parser.add_argument('--tensorboard',  default=True, type=bool,
                    help='Log progress to TensorBoard')

parser.add_argument('--GPU_ID', default=3, type=int,
                    help='GPU_id')
parser.add_argument('--num_updates', default=1, type=int,
                    help='number of meta-train')
parser.add_argument('--r', default=1, type=int,
                    help='repeat No')
`

Best,
Fengchun
